libs/structwalk/walktype.go

```diff
@@ -98,25 +97,20 @@ func walkTypeStruct(path *structpath.PathNode, st reflect.Type, visit VisitTypeF
 		if sf.PkgPath != "" {
 			continue // unexported
 		}
-		tag := sf.Tag.Get("json")
+		node := structpath.NewStructField(path, sf.Tag, sf.Name)
 
 		// Handle embedded structs (anonymous fields without json tags)
-		if sf.Anonymous && tag == "" {
+		if sf.Anonymous && node.JSONTag() == "" {
 			// For embedded structs, walk the embedded type at the current path level
 			// This flattens the embedded struct's fields into the parent struct
 			walkTypeValue(path, sf.Type, visit, visitedCount)
 			continue
 		}
 
-		if tag == "-" {
```

Comment:
I removed this because this seems like a typo. No reason we should be comparing here against the full tag.